File: NetFramework/GanttView/CS/CustomTaskInfoDialog/TaskInfoDialog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using C1.Win.C1GanttView;

namespace CustomTaskInfoDialog
{
    public partial class TaskInfoDialog : Form
    {
        #region field

        private C1GanttView _ganttView;
        private Task _selectedTask;

        #endregion

        #region ctors

        public TaskInfoDialog()
        {
            InitializeComponent();
        }

        public TaskInfoDialog(C1GanttView ganttView)
            : this()
        {
            _ganttView = ganttView;
        }

        #endregion

        #region event handlers

        private void TaskInfoDialog_Load(object sender, EventArgs e)
        {
            List<Task> tasks = new List<Task>();
            TaskCollection taskColl = _ganttView.Tasks;
            foreach (Task task in taskColl)
            {
                if (!string.IsNullOrEmpty(task.Name))
                {
                    tasks.Add(task);
                }
            }
            cbTask.DataSource = tasks;
            cbTask.DisplayMember = "Name";
            cbTask.SelectedItem = _ganttView.SelectedTask;
        }

        private void cbTask_SelectedIndexChanged(object sender, EventArgs e)
        {
            _selectedTask = (Task)cbTask.SelectedValue;
            tbTaskName.Text = _selectedTask.Name;
            tbResource.Text = _selectedTask.ResourceNames;
            tbStart.Text = _selectedTask.Start.Value.ToShortDateString();
            tbDuration.Text = _selectedTask.GetDurationInDays() + " day(s)";
            tbPercentComplete.Text = (_selectedTask.PercentComplete * 100).ToString("0.00");
        }

        #endregion

    }
}

File: NetFramework/GanttView/VB/CustomTaskInfoDialog/TaskInfoDialog.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.GanttView

Public Partial Class TaskInfoDialog
	Inherits Form
	#Region "field"

	Private _ganttView As C1GanttView
	Private _selectedTask As Task

	#End Region

	#Region "ctors"

	Public Sub New()
		InitializeComponent()
	End Sub

	Public Sub New(ganttView As C1GanttView)
		Me.New()
		_ganttView = ganttView
	End Sub

	#End Region

	#Region "event handlers"

	Private Sub TaskInfoDialog_Load(sender As Object, e As EventArgs)
		Dim tasks As New List(Of Task)()
		Dim taskColl As TaskCollection = _ganttView.Tasks
		For Each task As Task In taskColl
			If Not String.IsNullOrEmpty(task.Name) Then
				tasks.Add(task)
			End If
		Next
		cbTask.DataSource = tasks
		cbTask.DisplayMember = "Name"
		cbTask.SelectedItem = _ganttView.SelectedTask
	End Sub

	Private Sub cbTask_SelectedIndexChanged(sender As Object, e As EventArgs)
		_selectedTask = DirectCast(cbTask.SelectedValue, Task)
		tbTaskName.Text = _selectedTask.Name
		tbResource.Text = _selectedTask.ResourceNames
		tbStart.Text = _selectedTask.Start.Value.ToShortDateString()
		tbDuration.Text = _selectedTask.GetDurationInDays() & " day(s)"
		tbPercentComplete.Text = (_selectedTask.PercentComplete * 100).ToString("0.00")
	End Sub

	#End Region

End Class
